webapp/sources/rudder/rudder-web/src/main/scala/bootstrap/liftweb/checks/CheckNcfTechniqueUpdate.scala

```diff
@@ -90,104 +80,48 @@ class CheckNcfTechniqueUpdate(
-        methods            <- (parse(techniquesResult) \ "data" \ "generic_methods" match {
-                                case JObject(fields) =>
-                                  restExtractor.extractGenericMethod(JArray(fields.map(_.value))).map(_.map(m => (m.id,m)).toMap) match {
-                                    case Full(m) => Right(m)
-                                    case eb : EmptyBox =>
-                                      val fail = eb ?~!  s"An Error occured while extracting data from generic methods ncf API"
-                                      Left(JsonExtractionError(fail.messageChain, None))
-                                  }
-                                case a => Left(JsonExtractionError(s"Could not extract methods from ncf api, expecting an object got: ${a}", None))
-                              })
-
-
-        techniques         <- (parse(techniquesResult) \ "data" \ "techniques" match {
-                                case JObject(fields) =>
-                                  sequence(fields.map(_.value))(restExtractor.extractNcfTechnique(_, methods, false)) match {
-                                    case Full(m) => Right(m)
-                                    case eb : EmptyBox =>
-                                      val fail = eb ?~!  s"An Error occured while extracting data from techniques ncf API"
-                                      Left(JsonExtractionError(fail.messageChain, None))
+    import zio.syntax._
 
-                                  }
-                                case a => Left(JsonExtractionError(s"Could not extract techniques from ncf api, expecting an object got: ${a}", None))
-                              })
+    def updateNcfTechniques  = {
+      import NcfTechniqueUpgradeError._
+      for {
+        methods    <- techniqueReader.updateMethodsMetadataFile
+        techniques <- techniqueReader.updateTechniquesMetadataFile
         // Actually write techniques
-        techniqueUpdated   <- (sequence(techniques)(
+        written    <- ZIO.foreach(techniques)(
                                 techniqueWrite.writeTechnique(_, methods, ModificationId(uuidGen.newUuid), EventActor(systemApiToken.name.value)).toBox
-                              )) match {
-                                case Full(m) => Right(m)
-                                case eb : EmptyBox =>
-                                  val fail = eb ?~! "An error occured while writing ncf Techniques"
-                                  Left(WriteTechniqueError(fail.messageChain, None))
-                              }
-        // Update technique library once all technique are updated
-        libUpdate          <- techLibUpdate.update(ModificationId(uuidGen.newUuid), EventActor(systemApiToken.name.value), Some(s"Update Technique library after updating all techniques at start up")) match {
-                                case Full(techniques) => Right(techniques)
+                                  match {
```

Comment:
you need to `toIO` the box to correctly suspend `IO` for when the will be evaluated. 
You can also just `chainError` appart if you want to keep `WriteTechniqueError` type, in which case you can `mapError(err => WriteTechniqueError(err.fullMsg, None))`